### ISSUE TITLE
Add highlight in mode and progress sections

### ISSUE THREAD

**Issue by laraib07** [tamton-aquib/staline.nvim#7]
Liked your work.

Please consider adding highlight ( make font bold ) in mode and progress sections.

**Comment by tamton-aquib** [comment]
Heyyo @laraib07, sorry for the late response haha 😅.

And yes, adding highlights would be really nice. Will probably start soon. 
Also, I'm working on sections in the `beta` branch (just testing things out).

**Comment by laraib07** [comment]
Thank you for all your hard work on this 🙂.

**Comment by tamton-aquib** [comment]
heyyo @laraib07, I've merged beta to main.

Added sections. (wont break previous configs), but `filename_section` in `defaults` is deprecated.
Added the highlight stuff. give it as `{highlight_group, section_name}`.

Havent updated some stuff in readme. Will do it later.
In the mean time, feel free to try the new changes 😅

**Comment by laraib07** [comment]
Thanks for your work again 😄. 
I tried the new update. I believe fonts can be more bold. 
And there is another issue I am facing, don't know if its because of my configs. 
I am using edited Nvchad configs and previously used lualine. 
Now when using staline and setting laststatus=2, the statusline doesn't appear on startup, I have to toggle NvimTree twice (which behind the scene set laststatus to 0 and then to 2) to make it appears on the screen.

I haven't faced any issue with lualine on same configs.

**Comment by tamton-aquib** [comment]
Hey @laraib07, sorry for all these silly bugs.

> I am using edited Nvchad configs and previously used lualine.
> Now when using staline and setting laststatus=2, the statusline doesn't appear on startup, I have to toggle NvimTree twice (which behind the scene set laststatus to 0 and then to 2) to make it appears on the screen

The problem was, I didn't account for the lazy loading stuff that everybody is doing nowadays. 
So I've added the BufReadPost autocmd in the latest commit. Now it will show up when opening a normal buffer. (without extra performance cost)


> I believe fonts can be more bold.

have u tried setting `defaults = { font_active = "bold" }`. or maybe `"bold,italic"`
because by default it is `"none"`.

**Comment by laraib07** [comment]
> have u tried setting defaults = { font_active = "bold" }. or maybe "bold,italic"
because by default it is "none".

Now its perfect 😄😄. 
Thanks

**Comment by tamton-aquib** [comment]
Gosh.... I don't know how i miss these small things 😅

Fixed btw. And sorry for the inconveniences (again haha).

**Comment by laraib07** [comment]
I don't mind the inconveniences. I fully understand the effort goes behind these projects and I appreciate it.

Still facing the same issue
Branch name appears only after toggling nvimtree twice.

**Comment by tamton-aquib** [comment]
Added BufReadPost to `update_branch()` as well.

Also added some extra normal modes because `j` and `k` felt a little glitchy in NvChad. Probably because they are mapped to `gj` and `gk`, which enters an intermediate `normal` mode, i guess.

**Comment by laraib07** [comment]
The issue is resolved now🙂.
Will inform if I encounter any other bug.

**Comment by tamton-aquib** [comment]
Perfect.
And thanks again for reporting all these. I've actually seen some improvement in performance too.

Closing this issue for now. Feel free to open if there are any similar issues.
